src/main/java/com/example/shop26/service/OfferService.java

```diff
@@ -0,0 +1,94 @@
+package com.example.shop26.service;
+
+
+import com.example.shop26.model.Offer;
+import com.example.shop26.model.Price;
+import com.example.shop26.model.Product;
+import com.example.shop26.model.Quantity;
+import com.example.shop26.repository.OfferRepository;
+import com.example.shop26.repository.PriceRepository;
+import com.example.shop26.repository.ProductRepository;
+import com.example.shop26.repository.QuantityRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class OfferService {
+    private final OfferRepository offerRepository;
+    private final ProductRepository productRepository;
+    private final QuantityRepository quantityRepository;
+    private final PriceRepository priceRepository;
+
+
+    @Autowired

```

Comment:
Moyen fan de l'autowired , plus du bon vieux fichier de configuration mais why not

Reply:
idem à partir du moment où les configurations se complexifient un tant soit peu. Tu penses que pour juste 3/4 repository ca peut être overkill de faire un fichier de configuration, ou que ca passe en vrai ?

Reply:
Je te conseille de te renseigner un peu sur l'architecture hexagonale pour connaître la raison du commentaire au-dessus. Mais oui, il vaut mieux avoir l'habitude de passer par des classes de configuration